### ISSUE TITLE
Calling getClass.getSimpleName on objects returns name without trailing $

### ISSUE THREAD

**Issue by datalchemist** [scala-js/scala-js#3607]
There was some old closed issues about similar issue for getName #157 and #185 
But as I test it now with version 0.6.26, I don't get any trailing $ in object class name.
```
object Test
println(Test.getClass.getSimpleName)    /// result: Test
```

**Comment by sjrd** [comment]
I am not able to reproduce this issue. The following code compiles and prints `Test$`, not `Test`:
```scala
object Test

object HelloWorld {
  def main(args: Array[String]): Unit = {
    println(Test.getClass.getName)
  }
}
```
Can you provide more details about your environment? Things that come to mind:

* Is it an sbt project? A ScalaFiddle? Something else?
* What version of Scala?
* Provide the full source file. The snippet you posted above does not compile on its own as a `.scala` source file.
* Try the same code that you are using on Scala/JVM. Does it print something different?

**Comment by datalchemist** [comment]
Sorry, I did not take the time to test everything correctly. Actually, the problem concerns only getSimpleName and not getName. I wrongly assumed that both methods relied on the same logic.

Anyway, here is a snippet that returns a different result in jvm and js:
```
object TestClient {
  def testClient()=
    println("TestClient class: " + TestClient.getClass.getName  + " " +TestClient.getClass.getSimpleName)
}
```
On scala/jvm, it shows
> TestClient class: c.d.b.w.TestClient$ TestClient$

On scalajs, it shows
> TestClient class: c.d.b.w.TestClient$ TestClient

I encounter this problem because I am using a shared JSON formatter that relies on getSimpleName to encode case objects. Thus, on the jvm-backend, I am not able to decode the Json encoded in scalajs although I am using the same shared json formatter.

**Comment by sjrd** [comment]
Fixed in ffa6008f9862259783e6f2e012eca75996d2a41b.
